src/subdomains/core/liquidity-management/adapters/actions/base/ccxt-exchange.adapter.ts

```diff
@@ -115,7 +115,18 @@ export abstract class CcxtExchangeAdapter extends LiquidityActionAdapter {
     }
 
     try {
-      return await this.exchangeService.buy(tradeAsset, asset, amount);
+      try {
+        return await this.exchangeService.buy(tradeAsset, asset, amount);
+      } catch (e) {
+        if (e.message?.includes('not enough balance') && minTradeAmount != null) {
+          const availableBalance = await this.exchangeService.getBalance(tradeAsset);
+          if (availableBalance >= minTradeAmount) {
+            return await this.exchangeService.sell(tradeAsset, asset, availableBalance);
+          }
+        }
+
+        throw e;
+      }
```

Comment:
Hm I don't know, how this could work. Maybe you can do a proposition? The first catch should only be applied for the `buy` call, the second catch is for both `buy` and `sell`.

Reply:
Hm maybe I see something wrong 😅 I would only move the second try below, this should work too or?

I thought something like that:

`try {
      return await this.exchangeService.buy(tradeAsset, asset, amount);
    } catch (e) {
      try {
        if (e.message?.includes('not enough balance') && minTradeAmount != null) {
          const availableBalance = await this.exchangeService.getBalance(tradeAsset);
          if (availableBalance >= minTradeAmount) {
            return await this.exchangeService.sell(tradeAsset, asset, availableBalance);
          }
        }

        throw e;
      } catch (e) {
        if (e.message?.includes('not enough balance')) {
          throw new OrderNotProcessableException(e.message);
        }

        if (e.message?.includes('Illegal characters found')) {
          throw new Error(`Invalid trade request, tried to sell ${tradeAsset} for ${amount} ${asset}: ${e.message}`);
        }

        throw e;
      }
    }`